src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs

```diff
@@ -58,10 +68,29 @@ public FileTextLoader(string path, Encoding? defaultEncoding)
 
         internal sealed override string FilePath => Path;
 
+        /// <summary>
+        /// Creates <see cref="SourceText"/> from <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">Stream.</param>
+        /// <param name="workspace">Obsolete. Dummy <see cref="Workspace"/> instance.</param>
+        [Obsolete("Use CreateText(Stream, CancellationToken)")]
         protected virtual SourceText CreateText(Stream stream, Workspace workspace)
+            => _textFactory?.CreateText(stream, DefaultEncoding, CancellationToken.None) ?? EncodedStringText.Create(stream, DefaultEncoding);
+
+        /// <summary>
+        /// Creates <see cref="SourceText"/> from <see cref="Stream"/>.
+        /// </summary>
+        public virtual SourceText CreateText(Stream stream, CancellationToken cancellationToken)
+            => s_dummySourceText;
+
+        private SourceText CreateTextImpl(Stream stream, CancellationToken cancellationToken)
         {
-            var factory = workspace.Services.GetRequiredService<ITextFactoryService>();
-            return factory.CreateText(stream, DefaultEncoding);
+            var text = CreateText(stream, cancellationToken);
+
+            // Call the obsolete API for backwards compat if the new API is not overridden.
+#pragma warning disable CS0618 // Type or member is obsolete
+            return ReferenceEquals(text, s_dummySourceText) ? CreateText(stream, DummyWorkspace) : text;
```

Comment:
i'm missing where this DummyWorkspace is defined.

Reply:
found it.

Reply:
on the base class (TextLoader)